lib/delegate.rb

```diff
@@ -107,19 +107,17 @@ def respond_to_missing?(m, include_private)
     r
   end
 
-  KERNEL_RESPOND_TO = ::Kernel.instance_method(:respond_to?)
-  private_constant :KERNEL_RESPOND_TO
```

Comment:
Would it be enough to `Ractor.make_shareable(KERNEL_RESPOND_TO)` here?

Reply:
I think that unbound methods can't be made shareable right now. When I use `make_shareable` on `KERNEL_RESPOND_TO`, I get an error. So I think a workaround is needed for now.

Reply:
Ah indeed:
```
irb(main):001:0> KERNEL_RESPOND_TO = ::Kernel.instance_method(:respond_to?)
=> #<UnboundMethod: Kernel#respond_to?(*)>
irb(main):002:0> Ractor.make_shareable KERNEL_RESPOND_TO
<internal:ractor>:816:in `make_shareable': can not make shareable object for #<UnboundMethod: Kernel#respond_to?(*)> (Ractor::Error)
        from (irb):2:in `<main>'                                              
        from /home/eregon/.rubies/ruby-3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /home/eregon/.rubies/ruby-3.1.2/bin/irb:25:in `load'             
        from /home/eregon/.rubies/ruby-3.1.2/bin/irb:25:in `<main>'     
```
I think it's worth reporting to https://bugs.ruby-lang.org/

Reply:
cc @ko1